diff -urN a/drivers/soundwire/amd_init.c b/drivers/soundwire/amd_init.c
--- a/drivers/soundwire/amd_init.c	2026-02-19 07:31:37.000000000 -0800
+++ b/drivers/soundwire/amd_init.c	2026-03-02 13:54:40.526801116 -0800
@@ -157,7 +157,8 @@
 		if (!(ctx->link_mask & BIT(i)))
 			continue;
 		amd_manager = dev_get_drvdata(&ctx->pdev[i]->dev);
-		ret = amd_sdw_manager_start(amd_manager);
+		if (amd_manager)
+			ret = amd_sdw_manager_start(amd_manager);
 		if (ret)
 			return ret;
 	}
diff -urN a/drivers/soundwire/amd_manager.c b/drivers/soundwire/amd_manager.c
--- a/drivers/soundwire/amd_manager.c	2026-02-19 07:31:37.000000000 -0800
+++ b/drivers/soundwire/amd_manager.c	2026-03-02 13:54:40.526692948 -0800
@@ -485,6 +485,8 @@
 
 	dev_dbg(amd_manager->dev, "p_params->num:0x%x\n", p_params->num);
 	switch (amd_manager->acp_rev) {
+	case ACP60_PCI_REV:
+	case ACP6F_PCI_REV:
 	case ACP63_PCI_REV_ID:
 		switch (amd_manager->instance) {
 		case ACP_SDW0:
@@ -528,6 +530,8 @@
 	u32 offset_reg, lane_ctrl_ch_en_reg;
 
 	switch (amd_manager->acp_rev) {
+	case ACP60_PCI_REV:
+	case ACP6F_PCI_REV:
 	case ACP63_PCI_REV_ID:
 		switch (amd_manager->instance) {
 		case ACP_SDW0:
@@ -600,6 +604,8 @@
 	u32 lane_ctrl_ch_en_reg;
 
 	switch (amd_manager->acp_rev) {
+	case ACP60_PCI_REV:
+	case ACP6F_PCI_REV:
 	case ACP63_PCI_REV_ID:
 		switch (amd_manager->instance) {
 		case ACP_SDW0:
@@ -1025,6 +1031,8 @@
 	amd_manager->bus.controller_id = 0;
 	dev_dbg(dev, "acp_rev:0x%x\n", amd_manager->acp_rev);
 	switch (amd_manager->acp_rev) {
+	case ACP60_PCI_REV:
+	case ACP6F_PCI_REV:
 	case ACP63_PCI_REV_ID:
 		switch (amd_manager->instance) {
 		case ACP_SDW0:
diff -urN a/include/linux/soundwire/sdw_amd.h b/include/linux/soundwire/sdw_amd.h
--- a/include/linux/soundwire/sdw_amd.h	2026-02-19 07:31:37.000000000 -0800
+++ b/include/linux/soundwire/sdw_amd.h	2026-03-02 13:54:40.526528457 -0800
@@ -27,6 +27,8 @@
 #define ACP_SDW0	0
 #define ACP_SDW1	1
 #define AMD_SDW_MAX_MANAGER_COUNT	2
+#define ACP60_PCI_REV			0x60
+#define ACP6F_PCI_REV			0x6f
 #define ACP63_PCI_REV_ID		0x63
 #define ACP70_PCI_REV_ID		0x70
 #define ACP71_PCI_REV_ID		0x71
diff -urN a/sound/soc/amd/acp/acp-sdw-legacy-mach.c b/sound/soc/amd/acp/acp-sdw-legacy-mach.c
--- a/sound/soc/amd/acp/acp-sdw-legacy-mach.c	2026-02-19 07:31:37.000000000 -0800
+++ b/sound/soc/amd/acp/acp-sdw-legacy-mach.c	2026-03-02 13:54:40.528311620 -0800
@@ -166,6 +166,8 @@
 		}
 
 		switch (amd_ctx->acp_rev) {
+		case ACP60_PCI_REV:
+		case ACP6F_PCI_REV:
 		case ACP63_PCI_REV:
 			ret = get_acp63_cpu_pin_id(ffs(soc_end->link_mask - 1),
 						   *be_id, &cpu_pin_id, dev);
@@ -278,6 +280,8 @@
 		return -ENOMEM;
 
 	switch (amd_ctx->acp_rev) {
+	case ACP60_PCI_REV:
+	case ACP6F_PCI_REV:
 	case ACP63_PCI_REV:
 	case ACP70_PCI_REV:
 	case ACP71_PCI_REV:
@@ -326,6 +330,8 @@
 		return -ENOMEM;
 
 	switch (amd_ctx->acp_rev) {
+	case ACP60_PCI_REV:
+	case ACP6F_PCI_REV:
 	case ACP63_PCI_REV:
 	case ACP70_PCI_REV:
 	case ACP71_PCI_REV:
@@ -358,12 +364,11 @@
 	int sdw_be_num = 0, dmic_num = 0;
 	struct asoc_sdw_mc_private *ctx = snd_soc_card_get_drvdata(card);
 	struct snd_soc_acpi_mach_params *mach_params = &mach->mach_params;
-	struct asoc_sdw_endpoint *soc_ends __free(kfree) = NULL;
-	struct asoc_sdw_dailink *soc_dais __free(kfree) = NULL;
 	struct snd_soc_codec_conf *codec_conf;
 	struct snd_soc_dai_link *dai_links;
 	int num_devs = 0;
 	int num_ends = 0;
+	int num_confs;
 	int num_links;
 	int be_id = 0;
 	int ret;
@@ -374,17 +379,21 @@
 		return ret;
 	}
 
+	num_confs = num_ends;
+
 	/* One per DAI link, worst case is a DAI link for every endpoint */
-	soc_dais = kcalloc(num_ends, sizeof(*soc_dais), GFP_KERNEL);
+	struct asoc_sdw_dailink *soc_dais __free(kfree) =
+		kcalloc(num_ends, sizeof(*soc_dais), GFP_KERNEL);
 	if (!soc_dais)
 		return -ENOMEM;
 
 	/* One per endpoint, ie. each DAI on each codec/amp */
-	soc_ends = kcalloc(num_ends, sizeof(*soc_ends), GFP_KERNEL);
+	struct asoc_sdw_endpoint *soc_ends __free(kfree) =
+		kcalloc(num_ends, sizeof(*soc_ends), GFP_KERNEL);
 	if (!soc_ends)
 		return -ENOMEM;
 
-	ret = asoc_sdw_parse_sdw_endpoints(card, soc_dais, soc_ends, &num_devs);
+	ret = asoc_sdw_parse_sdw_endpoints(card, soc_dais, soc_ends, &num_confs);
 	if (ret < 0)
 		return ret;
 
@@ -396,7 +405,7 @@
 
 	dev_dbg(dev, "sdw %d, dmic %d", sdw_be_num, dmic_num);
 
-	codec_conf = devm_kcalloc(dev, num_devs, sizeof(*codec_conf), GFP_KERNEL);
+	codec_conf = devm_kcalloc(dev, num_confs, sizeof(*codec_conf), GFP_KERNEL);
 	if (!codec_conf)
 		return -ENOMEM;
 
@@ -407,7 +416,7 @@
 		return -ENOMEM;
 
 	card->codec_conf = codec_conf;
-	card->num_configs = num_devs;
+	card->num_configs = num_confs;
 	card->dai_link = dai_links;
 	card->num_links = num_links;
 
@@ -463,6 +472,10 @@
 	card->late_probe = asoc_sdw_card_late_probe;
 
 	snd_soc_card_set_drvdata(card, ctx);
+	if (mach->mach_params.subsystem_id_set)
+		snd_soc_card_set_pci_ssid(card,
+					  mach->mach_params.subsystem_vendor,
+					  mach->mach_params.subsystem_device);
 
 	dmi_check_system(soc_sdw_quirk_table);
 
diff -urN a/sound/soc/amd/acp/amd-acp63-acpi-match.c b/sound/soc/amd/acp/amd-acp63-acpi-match.c
--- a/sound/soc/amd/acp/amd-acp63-acpi-match.c	2026-02-19 07:31:37.000000000 -0800
+++ b/sound/soc/amd/acp/amd-acp63-acpi-match.c	2026-03-02 13:54:40.528426471 -0800
@@ -112,6 +112,30 @@
 	{}
 };
 
+static const struct snd_soc_acpi_adr_device rt1316_only_group_adr[] = {
+	{
+		.adr = 0x000030025D131601ull,
+		.num_endpoints = 1,
+		.endpoints = &spk_l_endpoint,
+		.name_prefix = "rt1316-1"
+	},
+	{
+		.adr = 0x000031025D131601ull,
+		.num_endpoints = 1,
+		.endpoints = &spk_r_endpoint,
+		.name_prefix = "rt1316-2"
+	},
+};
+
+static const struct snd_soc_acpi_link_adr acp63_rt1316_only[] = {
+	{
+		.mask = BIT(0),
+		.num_adr = ARRAY_SIZE(rt1316_only_group_adr),
+		.adr_d = rt1316_only_group_adr,
+	},
+	{}
+};
+
 struct snd_soc_acpi_mach snd_soc_acpi_amd_acp63_sof_sdw_machines[] = {
 	{
 		.link_mask = BIT(0) | BIT(1),
@@ -127,6 +151,11 @@
 struct snd_soc_acpi_mach snd_soc_acpi_amd_acp63_sdw_machines[] = {
 	{
 		.link_mask = BIT(0),
+		.links = acp63_rt1316_only,
+		.drv_name = "amd_sdw",
+	},
+	{
+		.link_mask = BIT(0),
 		.links = acp63_rt722_only,
 		.drv_name = "amd_sdw",
 	},
diff -urN a/sound/soc/amd/acp/amd-sdw-acpi.c b/sound/soc/amd/acp/amd-sdw-acpi.c
--- a/sound/soc/amd/acp/amd-sdw-acpi.c	2026-02-19 07:31:37.000000000 -0800
+++ b/sound/soc/amd/acp/amd-sdw-acpi.c	2026-03-02 13:54:40.530029785 -0800
@@ -36,6 +36,11 @@
 	ret = fwnode_property_read_u32_array(acpi_fwnode_handle(adev),
 					     "mipi-sdw-manager-list", &sdw_bitmap, 1);
 	if (ret) {
+		/* Try deprecated property name used by older BIOS */
+		ret = fwnode_property_read_u32_array(acpi_fwnode_handle(adev),
+						     "mipi-sdw-master-list", &sdw_bitmap, 1);
+	}
+	if (ret) {
 		dev_err(&adev->dev,
 			"Failed to read mipi-sdw-manager-list: %d\n", ret);
 		return -EINVAL;
diff -urN a/sound/soc/amd/acp/soc_amd_sdw_common.h b/sound/soc/amd/acp/soc_amd_sdw_common.h
--- a/sound/soc/amd/acp/soc_amd_sdw_common.h	2026-02-19 07:31:37.000000000 -0800
+++ b/sound/soc/amd/acp/soc_amd_sdw_common.h	2026-03-02 13:54:40.528470722 -0800
@@ -18,6 +18,8 @@
 #define ACP63_SDW_MAX_LINKS		2
 
 #define AMD_SDW_MAX_GROUPS		9
+#define ACP60_PCI_REV			0x60
+#define ACP6F_PCI_REV			0x6f
 #define ACP63_PCI_REV			0x63
 #define ACP70_PCI_REV			0x70
 #define ACP71_PCI_REV			0x71
diff -urN a/sound/soc/amd/ps/acp63.h b/sound/soc/amd/ps/acp63.h
--- a/sound/soc/amd/ps/acp63.h	2026-02-19 07:31:37.000000000 -0800
+++ b/sound/soc/amd/ps/acp63.h	2026-03-02 13:54:40.524695542 -0800
@@ -61,6 +61,9 @@
 
 #define ACP63_DMIC_ADDR			2
 #define ACP63_SDW_ADDR			5
+#define ACP60_PCI_REV			0x60
+#define ACP6F_PCI_REV			0x6f
+#define ACP60_SDW_ADDR			2
 #define AMD_SDW_MAX_MANAGERS		2
 
 /* time in ms for acp timeout */
@@ -370,6 +373,8 @@
 	u32 addr;
 	u32 reg_range;
 	u32 acp_rev;
+	u32 subsystem_vendor;
+	u32 subsystem_device;
 	u32 acp_sw_pad_keeper_en;
 	u32 acp_pad_pulldown_ctrl;
 	u16 acp63_sdw0_dma_intr_stat[ACP63_SDW0_DMA_MAX_STREAMS];
diff -urN a/sound/soc/amd/ps/pci-ps.c b/sound/soc/amd/ps/pci-ps.c
--- a/sound/soc/amd/ps/pci-ps.c	2026-02-19 07:31:37.000000000 -0800
+++ b/sound/soc/amd/ps/pci-ps.c	2026-03-02 13:54:40.524808178 -0800
@@ -112,6 +112,8 @@
 					break;
 				}
 				switch (adata->acp_rev) {
+				case ACP60_PCI_REV:
+				case ACP6F_PCI_REV:
 				case ACP63_PCI_REV:
 					adata->acp63_sdw0_dma_intr_stat[stream_id] = 1;
 					break;
@@ -126,6 +128,8 @@
 		}
 	}
 	switch (adata->acp_rev) {
+	case ACP60_PCI_REV:
+	case ACP6F_PCI_REV:
 	case ACP63_PCI_REV:
 		if (ext_intr_stat1 & ACP63_P1_AUDIO1_RX_THRESHOLD) {
 			writel(ACP63_P1_AUDIO1_RX_THRESHOLD,
@@ -335,6 +339,12 @@
 			mach->mach_params.links = mach->links;
 			mach->mach_params.link_mask = mach->link_mask;
 			mach->mach_params.subsystem_rev = acp_data->acp_rev;
+			mach->mach_params.subsystem_vendor = acp_data->subsystem_vendor;
+			mach->mach_params.subsystem_device = acp_data->subsystem_device;
+			mach->mach_params.subsystem_id_set = true;
+
+			dev_dbg(dev, "SSID %x%x\n", mach->mach_params.subsystem_vendor,
+				mach->mach_params.subsystem_device);
 			return mach;
 		}
 	}
@@ -430,8 +440,21 @@
 	if (dmic_en && wov_en)
 		is_dmic_dev = true;
 
+	/*
+	 * HP Dragonfly Pro (ACP6.0) firmware exposes SoundWire ACPI children but
+	 * does not mark the internal DMIC node with acp-audio-device-type.
+	 * Force-enable PDM so the machine driver can register DMIC capture.
+	 */
+	if (!is_dmic_dev &&
+	    acp_data->acp_rev == 0x60 &&
+	    acp_data->subsystem_vendor == 0x103c &&
+	    acp_data->subsystem_device == 0x8a7f)
+		is_dmic_dev = true;
+
 	if (acp_data->is_sdw_config) {
-		ret = acp_scan_sdw_devices(&pci->dev, ACP63_SDW_ADDR);
+		u64 sdw_addr = (acp_data->acp_rev < ACP63_PCI_REV) ?
+				ACP60_SDW_ADDR : ACP63_SDW_ADDR;
+		ret = acp_scan_sdw_devices(&pci->dev, sdw_addr);
 		if (!ret && acp_data->info.link_mask)
 			is_sdw_dev = true;
 	}
@@ -549,6 +572,8 @@
 		return -ENOMEM;
 
 	switch (adata->acp_rev) {
+	case ACP60_PCI_REV:
+	case ACP6F_PCI_REV:
 	case ACP63_PCI_REV:
 		acp63_hw_init_ops(adata->hw_ops);
 		break;
@@ -581,6 +606,8 @@
 
 	/* ACP PCI revision id check for ACP6.3, ACP7.0 & ACP7.1 platforms */
 	switch (pci->revision) {
+	case ACP60_PCI_REV:
+	case ACP6F_PCI_REV:
 	case ACP63_PCI_REV:
 	case ACP70_PCI_REV:
 	case ACP71_PCI_REV:
@@ -617,6 +644,9 @@
 	adata->addr = addr;
 	adata->reg_range = ACP63_REG_END - ACP63_REG_START;
 	adata->acp_rev = pci->revision;
+	adata->subsystem_vendor = pci->subsystem_vendor;
+	adata->subsystem_device = pci->subsystem_device;
+
 	pci_set_master(pci);
 	pci_set_drvdata(pci, adata);
 	mutex_init(&adata->acp_lock);
diff -urN a/sound/soc/amd/ps/ps-pdm-dma.c b/sound/soc/amd/ps/ps-pdm-dma.c
--- a/sound/soc/amd/ps/ps-pdm-dma.c	2026-02-19 07:31:37.000000000 -0800
+++ b/sound/soc/amd/ps/ps-pdm-dma.c	2026-03-02 13:54:40.531577789 -0800
@@ -352,6 +352,7 @@
 	.hw_params	= acp63_pdm_dma_hw_params,
 	.pointer	= acp63_pdm_dma_pointer,
 	.pcm_construct	= acp63_pdm_dma_new,
+	.use_dai_pcm_id = true,
 };
 
 static int acp63_pdm_audio_probe(struct platform_device *pdev)
diff -urN a/sound/soc/amd/ps/ps-sdw-dma.c b/sound/soc/amd/ps/ps-sdw-dma.c
--- a/sound/soc/amd/ps/ps-sdw-dma.c	2026-02-19 07:31:37.000000000 -0800
+++ b/sound/soc/amd/ps/ps-sdw-dma.c	2026-03-02 13:54:40.524902170 -0800
@@ -247,6 +247,8 @@
 	u32 sdw_mem_window_offset;
 
 	switch (acp_rev) {
+	case ACP60_PCI_REV:
+	case ACP6F_PCI_REV:
 	case ACP63_PCI_REV:
 		switch (manager_instance) {
 		case ACP_SDW0:
@@ -360,6 +362,8 @@
 		return -EINVAL;
 	stream_id = stream->stream_id;
 	switch (sdw_data->acp_rev) {
+	case ACP60_PCI_REV:
+	case ACP6F_PCI_REV:
 	case ACP63_PCI_REV:
 		switch (stream->instance) {
 		case ACP_SDW0:
@@ -437,6 +441,8 @@
 
 	byte_count.bytescount = 0;
 	switch (acp_rev) {
+	case ACP60_PCI_REV:
+	case ACP6F_PCI_REV:
 	case ACP63_PCI_REV:
 		switch (stream->instance) {
 		case ACP_SDW0:
@@ -518,6 +524,8 @@
 	if (!stream)
 		return -EINVAL;
 	switch (sdw_data->acp_rev) {
+	case ACP60_PCI_REV:
+	case ACP6F_PCI_REV:
 	case ACP63_PCI_REV:
 		switch (stream->instance) {
 		case ACP_SDW0:
@@ -564,6 +572,8 @@
 	stream = substream->runtime->private_data;
 	stream_id = stream->stream_id;
 	switch (acp_rev) {
+	case ACP60_PCI_REV:
+	case ACP6F_PCI_REV:
 	case ACP63_PCI_REV:
 		switch (stream->instance) {
 		case ACP_SDW0:
@@ -777,7 +787,7 @@
 	struct sdw_dma_dev_data *sdw_data;
 
 	sdw_data = dev_get_drvdata(dev);
-	if (sdw_data->acp_rev == ACP63_PCI_REV)
+	if (sdw_data->acp_rev <= ACP63_PCI_REV)
 		return acp63_restore_sdw_dma_config(sdw_data);
 	else
 		return acp70_restore_sdw_dma_config(sdw_data);
diff -urN a/sound/soc/amd/yc/acp6x.h b/sound/soc/amd/yc/acp6x.h
--- a/sound/soc/amd/yc/acp6x.h	2026-02-19 07:31:37.000000000 -0800
+++ b/sound/soc/amd/yc/acp6x.h	2026-03-02 13:54:40.532506328 -0800
@@ -14,6 +14,8 @@
 #define ACP6x_DEVS		3
 #define ACP6x_PDM_MODE		1
 
+#define ACP60_PCI_REV		0x60
+#define ACP6F_PCI_REV		0x6f
 #define ACP_SOFT_RESET_SOFTRESET_AUDDONE_MASK	0x00010001
 #define ACP_PGFSM_CNTL_POWER_ON_MASK	1
 #define ACP_PGFSM_CNTL_POWER_OFF_MASK	0
diff -urN a/sound/soc/amd/yc/pci-acp6x.c b/sound/soc/amd/yc/pci-acp6x.c
--- a/sound/soc/amd/yc/pci-acp6x.c	2026-02-19 07:31:37.000000000 -0800
+++ b/sound/soc/amd/yc/pci-acp6x.c	2026-03-02 13:54:40.524989029 -0800
@@ -6,6 +6,7 @@
  */
 
 #include <linux/pci.h>
+#include <linux/acpi.h>
 #include <linux/module.h>
 #include <linux/io.h>
 #include <linux/delay.h>
@@ -161,13 +162,20 @@
 
 	/* Yellow Carp device check */
 	switch (pci->revision) {
-	case 0x60:
-	case 0x6f:
+	case ACP60_PCI_REV:
+	case ACP6F_PCI_REV:
 		break;
 	default:
 		dev_dbg(&pci->dev, "acp6x pci device not found\n");
 		return -ENODEV;
 	}
+
+	/* Defer to pci-ps driver if SoundWire controller is described in ACPI */
+	if (ACPI_COMPANION(&pci->dev) &&
+	    acpi_find_child_device(ACPI_COMPANION(&pci->dev), 2, 0)) {
+		dev_dbg(&pci->dev, "SoundWire ACPI device found, deferring\n");
+		return -ENODEV;
+	}
 	if (pci_enable_device(pci)) {
 		dev_err(&pci->dev, "pci_enable_device failed\n");
 		return -ENODEV;
